gprofiler/main.py

```diff
@@ -1032,7 +1032,7 @@ def main() -> None:
                 else None
             )
         except APIError as e:
-            logger.error(f"Server error: {e}")
+            logger.critical(f"Server error: {e}")
```

Comment:
Wasn't the critical level applicable only for invalid API token? 
> At the same time we can change the "server error: invalid API key" log to be CRITICAL.
(see [this comment](https://github.com/Granulate/gprofiler/issues/676#issue-1575618054))

A known-message check might be needed here to trigger the critical level message in that specific case.
@Jongy , take a look?